OMCompiler/SimulationRuntime/c/util/omc_file.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * This file is part of OpenModelica.
+ *
+ * Copyright (c) 1998-CurrentYear, Open Source Modelica Consortium (OSMC),
+ * c/o Linköpings universitet, Department of Computer and Information Science,
+ * SE-58183 Linköping, Sweden.
+ *
+ * All rights reserved.
+ *
+ * THIS PROGRAM IS PROVIDED UNDER THE TERMS OF GPL VERSION 3 LICENSE OR
+ * THIS OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
+ * ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+ * RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3,
+ * ACCORDING TO RECIPIENTS CHOICE.
+ *
+ * The OpenModelica software and the Open Source Modelica
+ * Consortium (OSMC) Public License (OSMC-PL) are obtained
+ * from OSMC, either from the above address,
+ * from the URLs: http://www.ida.liu.se/projects/OpenModelica or
+ * http://www.openmodelica.org, and in the OpenModelica distribution.
+ * GNU version 3 is obtained from: http://www.gnu.org/copyleft/gpl.html.
+ *
+ * This program is distributed WITHOUT ANY WARRANTY; without
+ * even the implied warranty of  MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE, EXCEPT AS EXPRESSLY SET FORTH
+ * IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE CONDITIONS OF OSMC-PL.
+ *
+ * See the full OSMC Public License conditions for more details.
+ *
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "omc_file.h"
+
+FILE* omc_fopen(const char *filename, const char *mode)
+{
+#if defined(__MINGW32__) || defined(_MSC_VER)
+  MULTIBYTE_TO_WIDECHAR_LENGTH(filename, unicodeFilenameLength);
+  MULTIBYTE_TO_WIDECHAR_VAR(filename, unicodeFilename, unicodeFilenameLength);
+
+  MULTIBYTE_TO_WIDECHAR_LENGTH(mode, unicodeModeLength);
+  MULTIBYTE_TO_WIDECHAR_VAR(mode, unicodeMode, unicodeModeLength);
+
+  FILE *f = _wfopen(unicodeFilename, unicodeMode);
+
+  MULTIBYTE_OR_WIDECHAR_VAR_FREE(unicodeFilename);
+  MULTIBYTE_OR_WIDECHAR_VAR_FREE(unicodeMode);
+#else /* unix */
+  FILE *f = fopen(filename, mode);
+#endif
+  return f;
+}
+
+#if defined(__MINGW32__) || defined(_MSC_VER)
+int omc_stat(const char *filename, struct _stat *statbuf)
+{
+  MULTIBYTE_TO_WIDECHAR_LENGTH(filename, unicodeFilenameLength);
+  MULTIBYTE_TO_WIDECHAR_VAR(filename, unicodeFilename, unicodeFilenameLength);
+
+  int res;
+  res = _wstat(unicodeFilename, statbuf);
+
+  MULTIBYTE_OR_WIDECHAR_VAR_FREE(unicodeFilename);
+
+  return res;
+}
+#else /* unix */
+int omc_stat(const char *filename, struct stat *statbuf)
```

Comment:
Couldn't this just have been an inline function or a macro `omc_stat` to `stat`?

Reply:
Yeah we could make it a static inline function but I won't go for a macro. The windows version does the unicode conversions and the code will be very much non-readable with a macro.

Reply:
Only the pass-through function (unix) should be a macro. The windows function must stay a real function.

Reply:
Yeah fixing Unix code from Windows development environment is tricky. I will just push it in for now since release is near and will try to improve it later.